### PR TITLE
cleanup

### PR DESCRIPTION
**Pull request by lgarithm**

### CODE REVIEW

File: examples/c/alexnet.c [hunk 1]
```diff
@@ -5,11 +5,11 @@
...
 // https://papers.nips.cc/paper/4824-imagenet-classification-with-deep-convolutional-neural-networks.pdf
-s_model_t *alexnet(shape_t *image_shape, uint32_t arity)
+s_model_t *alexnet(const shape_t *image_shape, uint32_t arity)
 {
-    shape_ctx_t *sc = make_shape_ctx();
-    trait_ctx_t *tc = make_trait_ctx();
-    s_model_ctx_t *ctx = new_s_model_ctx();
+    shape_ctx_t *sc = new_shape_ctx();
+    trait_ctx_t *tc = new_trait_ctx();
+    s_model_ctx_t *ctx = make_s_model_ctx();
```
Comment (luomai) [inline examples/c/alexnet.c]: why shape and trait context are **new** and s_model context is **made**?

File: examples/c/alexnet.c [hunk 2]
```diff
@@ -56,16 +56,16 @@ s_model_t *alexnet(shape_t *image_shape, uint32_t arity)
...
-    free_s_layer(c1);
-    free_s_layer(c2);
-    free_s_layer(c3_c4);
-    free_s_layer(c5);
-    free_s_layer(f4096);
-    free_s_layer(f_out);
-    free_s_layer(pool);
-    free_s_layer(relu);
-    free_s_layer(out);
+    del_shape_ctx(sc);
```
Comment (luomai) [inline examples/c/alexnet.c]: deleting the layers is confusing to me. Actually we are release layer resources; but the layers, in logic, still exist in the model. The users shall still be able to query the model regarding layer statistics just like what we are doing in TL later.

File: examples/c/mnist_slp.c [hunk 3]
```diff
@@ -1,67 +1,41 @@
...
 
-    model_ctx_t *m = new_model_ctx();
-    node_t *x_ = make_placeholder(m, image_shape);
-    node_t *x = wrap_node(m, x_wrap_shape, x_);
-    node_t *w = make_parameter(m, weight_shape);
-    node_t *b = make_parameter(m, lable_shape);
+    symbol x = var(ctx, image_shape);
+    symbol x_ = reshape(ctx, mk_shape(sc, 1, shape_dim(image_shape)), x);
+    symbol w = covar(ctx, mk_shape(sc, 2, shape_dim(image_shape), arity));
+    symbol b = covar(ctx, mk_shape(sc, 1, arity));
```
Comment (luomai) [inline examples/c/mnist_slp.c]: @zsdonghao Is it easy for you to understand co-variable and variable?

In crystalnet, variable is equivalent to the placeholders in TF; while co-variable is equivalent to variables in TF.

File: src/crystalnet.h [hunk 4]
```diff
@@ -85,9 +67,9 @@ typedef struct s_node_t s_node_t;
 typedef s_node_t *symbol;
 typedef struct s_model_t s_model_t;
 typedef struct s_model_ctx_t s_model_ctx_t;
-extern s_model_ctx_t *new_s_model_ctx();
+extern s_model_ctx_t *make_s_model_ctx();
 extern s_model_t *new_s_model(s_model_ctx_t *, s_node_t *, s_node_t *);
-extern void free_s_model(s_model_t *);
+extern void del_s_model(s_model_t *);
```
Comment (luomai) [inline src/crystalnet.h]: if we only use s_model, shall we rename them all to model? 

The same applies to layer, node ...

File: examples/c/alexnet.c (hunk 1 above)
Comment (lgarithm) [reply examples/c/alexnet.c]: pointers obtained from `make_` functions are GCed automatically,
pointers obtained from `new_` functions needs to be freed via `del_` manually.

File: examples/c/alexnet.c (hunk 2 above)
Comment (lgarithm) [reply examples/c/alexnet.c]: Currently we don't have any logical layers saved in model, the layers are just macros or syntax sugars for constructing the model, they are useless once the model is constructed.

File: examples/c/mnist_slp.c (hunk 3 above)
Comment (lgarithm) [reply examples/c/mnist_slp.c]: In the future, complex models will be constructed using layer based APIs, there will be no `covar` any more.

File: src/crystalnet.h (hunk 4 above)
Comment (lgarithm) [reply src/crystalnet.h]: The physical structs, including model, layer, node etc will still exists internally.

**Pull request closed without merging**